src/modules/core/components/CountDownTimer/CountDownTimer.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { FormattedMessage, MessageDescriptor, defineMessage } from 'react-intl';
+
+import { MiniSpinnerLoader } from '~core/Preloaders';
+import QuestionMarkTooltip from '~core/QuestionMarkTooltip';
+
+import { useVotingExtensionParamsQuery, useBlockTimeQuery } from '~data/index';
+import { Address } from '~types/index';
+
+import styles from './CountDownTimer.css';
+import { calculateTimeLeft } from '~utils/time';
+
+const MSG = defineMessage({
+  days: {
+    id: 'CountDownTimer.CountDownTimer.days',
+    defaultMessage: ' {days}d',
+  },
+  hours: {
+    id: 'CountDownTimer.CountDownTimer.hours',
+    defaultMessage: ' {hours}h',
+  },
+  minutes: {
+    id: 'CountDownTimer.CountDownTimer.minutes',
+    defaultMessage: ' {minutes}m',
+  },
+  seconds: {
+    id: 'CountDownTimer.CountDownTimer.seconds',
+    defaultMessage: ' {seconds}s',
+  },
+  loadingText: {
+    id: 'CountDownTimer.CountDownTimer.loadingText',
+    defaultMessage: 'Loading countdown period',
+  },
+});
+
+interface Props {
+  text: MessageDescriptor | string;
+  periodType:
+    | 'stakePeriod'
+    | 'submitPeriod'
+    | 'revealPeriod'
+    | 'escalationPeriod';
+  createdAt: number;
+  colonyAddress: Address;
+  tooltipText?: MessageDescriptor | string;
+}
+
+const CountDownTimer = ({
+  text,
+  colonyAddress,
+  createdAt,
+  tooltipText,
+  periodType,
+}: Props) => {
+  const { data, error, loading } = useVotingExtensionParamsQuery({
+    variables: { colonyAddress },
+  });
+  const stakePeriod = data?.votingExtensionParams[periodType];
+  const { data: blockTimeData } = useBlockTimeQuery();
+
+  const differenceVsBCTime = useMemo(
+    () =>
+      blockTimeData?.blockTime ? blockTimeData?.blockTime - Date.now() : 0,
+    [blockTimeData],
+  );
+
+  const [timeLeft, setTimeLeft] = useState(
+    calculateTimeLeft(createdAt, differenceVsBCTime, stakePeriod),
+  );
+
+  useEffect(() => {
+    if (stakePeriod !== undefined) {
+      const timer = setInterval(() => {
+        setTimeLeft(
+          calculateTimeLeft(createdAt, differenceVsBCTime, stakePeriod),
+        );
+      }, 1000);
+      return () => clearInterval(timer);
+    }
+    return undefined;
+  }, [createdAt, stakePeriod, differenceVsBCTime]);
+
+  if (loading || data === undefined) {
+    return <MiniSpinnerLoader loadingText={MSG.loadingText} />;
+  }
```

Comment:
The bug where the timer is not displayed after loading is still there:

![bug-countdown-spinner-2](https://user-images.githubusercontent.com/1193222/114410725-450f9600-9bb4-11eb-8ddf-42feac042272.gif)

I think the only way to fix this properly is to store our own loading checks in react's state, and only setting that to false when `calculateTimeLeft` is done